### ISSUE TITLE
JackCtrl - Jack 1.9.17 doesn't start on Win10 64bit

### ISSUE THREAD

**Issue by alfonso73** [jackaudio/jack2#714]
Hi JackCtrl/Jack 1.9.17 freezes when launched on Windows 10 64bit

## Environment

* **JACK Version:** Which version of jack2 are you using? (e.g. the output of `jackd --version` or a git commit checksum)
* **Operating System:** What operating system or distribution in which version are you using? (e.g. Linux, macOS, Windows)
* **Installation:** How did you install jack2? (e.g. package manager, from source)

## Steps To Reproduce

```bash
# Paste a minimal code example here (e.g. about how you started JACK)
```

### Expected vs. actual behavior

A clear and concise description of what you expected to happen vs. what happened.

**Comment by falkTX** [comment]
please give some more description of the issue please..

**Comment by kmatheussen** [comment]
Does qjackctl freeze? Can you open the "messages" window before pressing
start, and post a screenshot when it freezes?

**Comment by alfonso73** [comment]
I just start qjackctl by double clicking on qjackctl.exe and the whole thing freezes forever and stays as you can see in the screenshot. Btw previous versions I've got used to work flawlessy (apart of the well known security related problem that Supercollider's server doesn't boot on Win10 64x)
![qjackctl_freeze](https://user-images.githubusercontent.com/3230434/105866830-b055bd00-5ff4-11eb-958e-b0edf97d9ead.png)

**Comment by alfonso73** [comment]
I suspect that it's still a security issue. I say this because after I installed the new Jack 1.9.17 I have a new suspicious symptom from SuperCollider because it's server refuses to start and kind of freezes the server booting process (with previous Jack versions it used to be the same but SC gave an error code). As soon as I uninstall Jack 1.9.17, SuperCollider works correctly again.
Excuse me if I use the SC<>Jack interaction problem as a proof of the "bug" (if it is a bug indeed), but it seems to me that it gives good clues of what's goin' on.

**Comment by kmatheussen** [comment]
Are you sure sure it's a bug in Jack? It seems like a bug in qjackctl. Can
you try to run this program to reset qjackctl settings?
http://users.notam02.no/~kjetism/qjackctl_reset.exe

**Comment by kmatheussen** [comment]
And if that didn't work, can you try this older version of qjacktl?
http://users.notam02.no/~kjetism/qjackctl.exe

**Comment by alfonso73** [comment]
Reseting qjackclt settings i have the same situation.
With the old qjackctl i have another variant of teh problem...but still freeze as you cna see in the screenshots
![Screenshot 2021-01-27 174421](https://user-images.githubusercontent.com/3230434/106023976-9afd9300-60c7-11eb-8632-42216c9e8501.png)
![Screenshot 2021-01-27 174508](https://user-images.githubusercontent.com/3230434/106023982-9c2ec000-60c7-11eb-838b-1ceddfa5682c.png)

**Comment by kmatheussen** [comment]
Does jack work if you run it from the windows console? Like this (I think):

cd C:\Program Files\JACK2
jackd -dportaudio

**Comment by alfonso73** [comment]
I have this...and no signs of jack running in the task manager.
![Screenshot 2021-01-27 225318](https://user-images.githubusercontent.com/3230434/106059359-e7aa9380-60f2-11eb-9e46-429a62af4775.png)

**Comment by kmatheussen** [comment]
From the screenshot, it looks like jack runs fine. You don't see jackd.exe
running under "Windows Command Processor" or something like that in the
task manager?

**Comment by kmatheussen** [comment]
@rncbc Does qjackctl access jack in heavy ways, or do any other things, during startup, that could explain the freeze?

**Comment by rncbc** [comment]
> Does qjackctl access jack in heavy ways, or do any other things, during startup, that could explain the freeze?

what are "heavy ways" ? qjackctl only access jack as a normal client and, if instructed so, spawns the jackd.exe process.

on windows it doesn't do the jackdbus thing, I believe.

**Comment by kmatheussen** [comment]
On Thu, Jan 28, 2021 at 6:59 PM Rui Nuno Capela <notifications@github.com>
wrote:

> Does qjackctl access jack in heavy ways, or do any other things, during
> startup, that could explain the freeze?
>
> what are "heavy ways" ? qjackctl only access jack as a normal client and,
> if instructed so, spawns the jackd.exe process.
>
Sorry, I'm not sure what I meant by "heavy ways". :-)
Does qjackctl call any libjack functions or any portaudio functions during
startup?

>

**Comment by rncbc** [comment]
> Does qjackctl call any libjack functions or any portaudio functions during startup?

of course it does call on libjack! duh? how could it be any other way?

re. portaudio, I really don't know for sure, but the portaudio discovery was integrated in qjackctl setup dialog by,... guess who? ;)

**Comment by kmatheussen** [comment]
On Thu, Jan 28, 2021 at 9:01 PM Rui Nuno Capela
<notifications@github.com> wrote:
>
> Does qjackctl call any libjack functions or any portaudio functions during startup?
>
> of course it does call on libjack! duh? how could it be any other way?
>

I'm not quite sure what you mean, I'm just trying to figure out how
both jackd and qjackctl freezes at startup, as reported here. Doesn't
qjackctl check if jack is already running when it starts up? Maybe
that call never finishes...

> re. portaudio, I really don't know for sure, but the portaudio discovery was integrated in qjackctl setup dialog by,... guess who? ;)
>
Actually I think it was sletz. :-)
(I just improved it to run in it's own thread to prevent the GUI from
freezing while scanning for devices)
But I guess it doesn't do this during startup.

**Comment by dyfer** [comment]
> I have this...and no signs of jack running in the task manager.

@alfonso73 This does look like jack has started.
When I start jack in the command prompt, in the Task Manager it shows up as a subprocess of `Windows Command Processor`. 
Once jack runs in the command prompt, can you start QJackCtl? For me it starts and shows that jack is running (I don't need to click "start" because jack is already started)

**Comment by alfonso73** [comment]
Hi, yes i can see the process where you suggested. But qjackctl still
refuses to start. (and Supercollider refuse to start its server unless i
unistall jack2)

**Comment by kmatheussen** [comment]
Can you try to run jack like this instead:

jackd.exe -ddummy

And tell us if you can start qjackctl or supercollider afterwards.

**Comment by dyfer** [comment]
@alfonso73 maybe you have the remnants of an old jack installation? Check if you don't have anything in `C:\Program Files (x86)\Jack\` (I think current install goes in `Program Files\JACK2`)

**Comment by yyf** [comment]
Hi, 
I'm experiencing the same issue on Windows 10. QjackCtl freezes at launch and logs below message. After googling a bit, I wasn't able to find related information regarding the BDB1546 error message. Any idea? 

"Could not connect to JACK server as client. -Overall operation failed. - Unable to connect to sever. Please check the messages window for more info. ... Jack is stopping... BDB1546 unable to join the environment... cannot open DB environment: Resource temporarily unavilable."

I also noticed Portaudio takes a long time probing audio hardware and eventually times out, when I try to select interface in the QjackCtl setup window. I double-checked there is no old version of Jack in my Program Files (x86) path, and can reproduce the same error message launching Jack in command line. 

Thanks.

**Comment by falkTX** [comment]
what is the output of:
```
jackd -d portaudio -l
```
supposedly it prints the list of devices.
there might be some certain devices that are causing this timeout.

The "BDB" stuff is related to meta-data. Weird with that error, but should not be what is causing this.

**Comment by yyf** [comment]
@falkTX here is the output I got: 

> jackdmp 1.9.17
Copyright 2001-2005 Paul Davis and others.
Copyright 2004-2016 Grame.
Copyright 2016-2021 Filipe Coelho.
jackdmp comes with ABSOLUTELY NO WARRANTY
This is free software, and you are welcome to redistribute it
under certain conditions; see the file COPYING for details
Drivers/internals found in : C:\Program Files\JACK2
Drivers/internals found in : C:\Program Files\JACK2
JACK server starting in realtime mode with priority 10
self-connect-mode is "Don't restrict self connect requests"
BDB1546 unable to join the environment
cannot open DB environment: Resource temporarily unavailable
BDB1546 unable to join the environment
cannot open DB environment: Resource temporarily unavailable
Cannot connect to named pipe after wait = \\.\pipe\server_jack_default_0 err = 2
Cannot connect to server request channel
jack server is not running or cannot be started
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
Open error: is JACK server running?
jack_deactivate called with a NULL client
jack_deactivate called with a NULL client
jack_client_close called with a NULL client
********************** Devices list, 27 detected **********************
(... audio device info ...)
**************************** End of list ****************************
Cannot initialize driver
JackServer::Open failed with -1
Failed to open server

**Comment by falkTX** [comment]
does that take a long time to run? like if it is timing out

**Comment by yyf** [comment]
It's not immediate but didn't time out.

**Comment by alfonso73** [comment]
with -ddummy i have the same behavior...no qjackctl no SC server starting.
I've no residuals of previous Jack neither in Program Files (x86) neither in the path in system enviroment variables.
With jackd -d portaudio -l
i have the output in the screenshot here
![Screenshot 2021-01-29 154459](https://user-images.githubusercontent.com/3230434/106289260-5ac61e00-6249-11eb-8a21-7690f6d33c31.png)

it seems to me that jack doesn't really start...

**Comment by falkTX** [comment]
please run this instead, which will use verbose mode and alternative server name. so it could give some more info:
```
jackd -v -n testing -d portaudio -l
```

**Comment by yyf** [comment]
@falkTX Here you go. Seems it failed to open server. Any suggestion? 

> C:\Program Files\JACK2>jackd -v -n testing -d portaudio -l
jackdmp 1.9.17
Copyright 2001-2005 Paul Davis and others.
Copyright 2004-2016 Grame.
Copyright 2016-2021 Filipe Coelho.
jackdmp comes with ABSOLUTELY NO WARRANTY
This is free software, and you are welcome to redistribute it
under certain conditions; see the file COPYING for details
Drivers/internals found in : C:\Program Files\JACK2
Drivers/internals found in : C:\Program Files\JACK2
JACK server starting in realtime mode with priority 10
self-connect-mode is "Don't restrict self connect requests"
Jack: JackWinThread::StartImp : create non RT thread
Jack: JackWinThread::ThreadHandler : start
Jack: Initializing PortAudio...
Cannot connect to named pipe after wait = \\.\pipe\server_jack_default_0 err = 2
Cannot connect to server request channel
jack server is not running or cannot be started
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
Open error: is JACK server running?
jack_deactivate called with a NULL client
jack_deactivate called with a NULL client
jack_client_close called with a NULL client
********************** Devices list, 23 detected **********************
... (audio device info) ...
**************************** End of list ****************************
Cannot initialize driver
Jack: no message buffer overruns
Jack: JackWinThread::Stop
Jack: JackWinThread::ThreadHandler : exit
JackServer::Open failed with -1
Jack: Succeeded in unlocking 107341338 byte memory area
Jack: JackShmMem::delete size = 0 index = 0
Jack: ~JackDriver
Jack: Succeeded in unlocking 1187 byte memory area
Jack: JackShmMem::delete size = 0 index = 1
Jack: Cleaning up shared memory
Jack: Cleaning up files
Jack: Unregistering server `testing'
Failed to open server

**Comment by alfonso73** [comment]
this is what i get

C:\Program Files\JACK2>jackd -v -n testing -d portaudio -l
jackdmp 1.9.17
Copyright 2001-2005 Paul Davis and others.
Copyright 2004-2016 Grame.
Copyright 2016-2021 Filipe Coelho.
jackdmp comes with ABSOLUTELY NO WARRANTY
This is free software, and you are welcome to redistribute it
under certain conditions; see the file COPYING for details
Drivers/internals found in : C:\Program Files\JACK2
Drivers/internals found in : C:\Program Files\JACK2
JACK server starting in realtime mode with priority 10
self-connect-mode is "Don't restrict self connect requests"

**Comment by alfonso73** [comment]
Some more strange symptoms.
A lot of different apps don't start after Jack 1.9.17 is installed (VCV Rack, Max8, NI Reaktor standalone, NI Kontakt standalone).
When Jack2 is uninstalled Max8 is back working again but VCV and NI stuff doesn't start no more.
I tried register/unregister jackRouter.dll but no improvements.
Probably uninstalling Jack leave some garbage around? Infact the JackRouter folder doesn't get deleted and i have to delete it manually. But anyway even if deleted VCV and NI stuff refuses to start.
Should i open a new issue? It sounds that it'is all related...
BTW never got this behavior for Jack version from several months ago. I just got the SC server issue but no other problems.

**Comment by alfonso73** [comment]
An update about some apps not starting even after JACK2 is unistalled.
I did a Windows registry cleaning process using a specific app and now all those apps works normally.
Waiting for some news about this issue and see if i can try install JACK2 again. Thanks and all the best

**Comment by dts350z** [comment]
I have the same problem same symptoms on windows 10. 

Jack1.9.17 (or the previous version) hangs on start.

Apps that check ASIO devices hang on start, until Jack2 is uninstalled. e.g. Plogue Bidule.

Microsoft Windows [Version 10.0.19042.746]
(c) 2020 Microsoft Corporation. All rights reserved.

C:\Users\Glenn.THEBEAST>"C:\Program Files\JACK2\jackd.exe" -v -n testing -d portaudio -l
jackdmp 1.9.17
Copyright 2001-2005 Paul Davis and others.
Copyright 2004-2016 Grame.
Copyright 2016-2021 Filipe Coelho.
jackdmp comes with ABSOLUTELY NO WARRANTY
This is free software, and you are welcome to redistribute it
under certain conditions; see the file COPYING for details
Drivers/internals found in : C:\Program Files\JACK2
Drivers/internals found in : C:\Program Files\JACK2
JACK server starting in realtime mode with priority 10
self-connect-mode is "Don't restrict self connect requests"
<------Hangs here

**Comment by SynthiaPayne** [comment]
Hello All,

Below are the steps that allowed me successful operation on Windows10 of:
JACK2 1.9.17 -> Jacktrip 1.3 -> Reaper (via JackRouter)

NOTICE: This was an experiment that worked for me. I don't know if it will work for you but it's worth a try!

Issue:
JACK2 suddenly began opening blank white windows and then hang

Workarounds:
1a) Cleared out all previous Jack including registry items
1b) Combined the 32-bit and 64-bit versions of JACK2 (my computer is 64)
1c) Moved the latest version of Jacktrip 1.3 into the JACK2 folder

Aaaaand we're off!

synth

"Go at it boldly, and you'll find unexpected forces closing round you and coming to your aid." -Basil King

**Comment by cliffordmanasseh** [comment]
Had a similar issue with jack audio not starting, and throwing an unable to start, sorry message always. working fine after installing the 32 bit version

**Comment by nyarla** [comment]
Hi, I maybe found workaround for this problem.

In my environment, Old version (1.9.16 and 1.9.15) of JackAudio stopped working after installed 1.9.17,
but I deleted `C:\Users\{username}\AppData\Local\Temp\jack_db` directory, I got back working fine of JackAudio.

And note, I tries these actions for fix about JackAudio does not works,
but that is not enable to fix of this problem:

- Uninstall and re-install JackAudio
- Delete JackAudio-related registory values
- Downgrade version of JackAudio

Finally, JackAudio 1.9.17 works fine in my environment after I deleted `C:\Users\{username}\AppData\Local\Temp\jack_db`,
I think that is workaround for fix about JackAudio does not launch from QJackCtl or windows console.

**Comment by christofmuc** [comment]
@nyarla Thank you! Deleting the jack_db in the AppData directory allowed both the jackd as well as the QJackCtl to finally launch!

**Comment by dts350z** [comment]
For me, QJackCtl will now start (after removing jack_db), and sometimes I can get the router to start without errors, however in ALL cases is still doesn't work with any applications. No channels are seen, can't initialize device, etc.

**Comment by manikandhan31** [comment]
> 
> 
> Hi, I maybe found workaround for this problem.
> 
> In my environment, Old version (1.9.16 and 1.9.15) of JackAudio stopped working after installed 1.9.17,
> but I deleted `C:\Users\{username}\AppData\Local\Temp\jack_db` directory, I got back working fine of JackAudio.
> 
> And note, I tries these actions for fix about JackAudio does not works,
> but that is not enable to fix of this problem:
> 
>     * Uninstall and re-install JackAudio
> 
>     * Delete JackAudio-related registory values
> 
>     * Downgrade version of JackAudio
> 
> 
> Finally, JackAudio 1.9.17 works fine in my environment after I deleted `C:\Users\{username}\AppData\Local\Temp\jack_db`,
> I think that is workaround for fix about JackAudio does not launch from QJackCtl or windows console.

Thanks, this worked for me after hours of struggling to bring up QjackCtl

**Comment by amurzeau** [comment]
I got the same issue and fixed it by removing via the uninstaller the old version.

I found that when having the issue, the newer jackd.exe was using `C:\WINDOWS\SYSWOW64\libjack.dll` or something like that. The version of jackd.exe is 1.9.17 but the DLL was the previous version (1.9.11 for me). (You can see which DLL jackd.exe is using, using process explorer)
The newer 32 bits jack install its DLL inside C:\WINDOWS\libjack.dll instead and this folder seems to have a lower priority over C:\WINDOWS\SYSWOW64.

**Comment by cloudcrypt** [comment]
I've been having different issues with Jack, as outlined in #706.
It seems to be that for me, an installation of Jack2 v1.9.17 64-bit would not work with any **32-bit applications** that are clients of JackRouter.

From what @amurzeau said above, I checked and did have `libjack.dll` in `C:\Windows\SysWOW64\`. And it does seem that the reason only the **32-bit** applications would break would be that `libjack.dll` (as the 32-bit lib) is being prioritized in `SysWOW64` over the one in C:\Windows. Since the one is `SysWOW64` is an old version and remnant of some 1.9.11 install, any application trying to use it breaks (with the CheckSize and CheckRead error shown in #706). 

Manually removing `C:\Windows\SysWOW64\libjack.dll` has solved all issues for my 64-bit Jack2 1.9.17 install. (Thank you @amurzeau!)

My theory for this issue is that at some point the 1.9.11 uninstaller did not remove `C:\Windows\SysWOW64\libjack.dll`, and as such, it got stuck there since, never being removed on any install/uninstall cycle of either 1.9.11 or 1.9.17.

**Comment by alfonso73** [comment]
just tried again on a clean Win10 machine
same problems about Qjackctl don't starting and i have this

C:\Program Files\JACK2>jackd -dportaudio
jackdmp 1.9.17
Copyright 2001-2005 Paul Davis and others.
Copyright 2004-2016 Grame.
Copyright 2016-2021 Filipe Coelho.
jackdmp comes with ABSOLUTELY NO WARRANTY
This is free software, and you are welcome to redistribute it
under certain conditions; see the file COPYING for details
Drivers/internals found in : C:\Program Files\JACK2
Drivers/internals found in : C:\Program Files\JACK2
JACK server starting in realtime mode with priority 10
self-connect-mode is "Don't restrict self connect requests"

**Comment by dyfer** [comment]
@alfonso73 what driver are you using with Jack? Have you tried different buffer sizes and different number of channels?

**Comment by alfonso73** [comment]
@dyfer i can't even access any control on QjackCtl because its GUI is totally blank when i launch it and is marked as "Not responding" in the Task Manager

**Comment by alfonso73** [comment]
I think i got it working now.

Deleted C:\Users{username}\AppData\Local\Temp\jack_db folder
Started QjackCtl that now after folder deleted at 1. works
Edit all the setup in QjackCtl selecting the ASIO driver, setting sample rate, latency etc.
Started SC using JackRouter.
Everything works

**Comment by dyfer** [comment]
Should this issue be closed then?

**Comment by alfonso73** [comment]
@dyfer i'm not so sure beacause when Jack 1.9.17 is installed doesn't work unless some tricky things on the user side are done. So probably for a working release i guess the issue has to be addressed.

**Comment by dyfer** [comment]
I believe jack_db folder is an issue that sometimes occurs if jack 1.9.11 was previously installed. Was that not the case?

**Comment by alfonso73** [comment]
@dyfer Nope. On a clean system that folder is created as soon as Jack start as far as i can see. That's why, i guess, has to be deleted before starting QjackCtl for the first time, probably because it contains some settings that doesn't fit the jack/audio driver combination.

**Comment by aldocisneros** [comment]
@chairbender 's fix has been working for me for a while now. I'm using a Behringer UMC404HD interface on windows 11, and have been running Jack (QjackCTL) without problems for a while now. In my case, I just needed to increase the delay time from 2 to 3 seconds.

Found their solution on @ghmechamanias 's issue (link below)

https://github.com/jackaudio/jack2/issues/779#issuecomment-1079842860

**Comment by cdhowie** [comment]
Came here from Google after having qjackctl freeze, and deleting the temporary `jack_db` was all it took to fix the problem. A huge thank you to @nyarla for saving hours of my time.

To add some data points to this issue, I am running 1.9.17 and have been for probably a year. I did not uninstall, install, or upgrade jack since then. This issue happened today spontaneously -- jack was working fine when I locked my computer to go to bed last night, and when I got up this morning jackd and qjackctl had frozen. I had to use the task manager to kill them, and then qjackctl refused to start normally, freezing before even rendering its UI.

So this is definitely not related to having earlier versions of jack installed.  Somehow 1.9.17 can get itself into this situation.

**Comment by diqidoq** [comment]
Didn't read all comments but the most means 60-70% down ... and I wonder HOW YOU EVEN install JACK on Windows correctly since it doesnt even has jackd as a registered command here on Win10 cmd to get informal output with latest Jack connection Kit installed (21) by simply double clikcing the installer exe from https://jackaudio.org/downloads/ Win 64bits

There seem to be a JackRouter in the registry and I can open qJackctl UI but cannot start the server (of course). How do you guys install Jack since Win documentation on it is rare.
